### ISSUE TITLE
Do not take into account provision resources that do not match the supplied teflo label

### ISSUE THREAD

**Issue by ryankwilliams** [RedHatQE/teflo#199]
When using teflo labels, teflo should only take into account resources matching that label when generating the `hosts` key for any orchestrate/execute task resources. The current behavior will include all provision resources that are apart of the group defined even if it does not have the matching label.

Example:

*Provision SDF*

```yaml
---
name: provision
description: Provision

provision:
  - name: scenario_driver
    groups:
      - localhost
      - scenario_drivers
    ip_address: 127.0.0.1
    ansible_params:
      ansible_connection: local
    labels:
      - stack-beaker-libvirt-only
      - stack-beaker-bare_metal-only

  - name: baremetal
    groups:
      - hypervisor <<<<
    provisioner: beaker-client
    <truncated>
    labels:
      - stack-beaker-libvirt-only

  - name: prod_test_driver
    groups:
      - prod_test_drivers
      - hypervisor <<<<
    provisioner: beaker-client
    <truncated>
    labels:
      - stack-beaker-bare_metal-only
```

When I issue the following teflo command below, the resulting result SDF file for orchestrate is generated.

```
teflo run -t validate -s SDF.yml -l stack-beaker-libvirt-only
```

```yaml
name: orchestrate
description: Orchestrate
remote_workspace: []
resource_check: {}
provision: []
orchestrate:
- name: Install OpenShift on libvirt
  description: null
  orchestrator: ansible
  hosts: <<<<
  - baremetal
  - prod_test_driver
  ansible_playbook:
    name: ansible/install_ocp_libvirt.yml
  ansible_options:
    extra_vars:
      artifact_directory: /home/rywillia/Projects/gitlab/mpqe/mps/solutions/cnv_odf_ocp/venv/cnv_odf_ocp/.teflo/.results/artifacts/sharedArtifacts
  ansible_galaxy_options:
    role_file: requirements.yml
  cleanup:
    name: Uninstall OpenShift on libvirt
    orchestrator: ansible
    hosts: hypervisor
    ansible_playbook:
      name: ansible/uninstall_ocp_libvirt.yml
    ansible_options:
      extra_vars:
        foo: bar
    ansible_galaxy_options:
      role_file: requirements.yml
    labels:
    - stack-beaker-libvirt-only
  labels:
  - stack-beaker-libvirt-only
```

This causes issues when ansible goes to run the playbook. The inventory file has a group named `hypervisor` where only one of the hosts in the group is valid and the other does not exist.

The behavior when using labels should not account the second resource that has the non matching label. This should be the expected result orchestrate SDF file.

```yaml
name: orchestrate
description: Orchestrate
remote_workspace: []
resource_check: {}
provision: []
orchestrate:
- name: Install OpenShift on libvirt
  description: null
  orchestrator: ansible
  hosts: <<<<
  - baremetal
  ansible_playbook:
    name: ansible/install_ocp_libvirt.yml
  ansible_options:
    extra_vars:
      artifact_directory: /home/rywillia/Projects/gitlab/mpqe/mps/solutions/cnv_odf_ocp/venv/cnv_odf_ocp/.teflo/.results/artifacts/sharedArtifacts
  ansible_galaxy_options:
    role_file: requirements.yml
  cleanup:
    name: Uninstall OpenShift on libvirt
    orchestrator: ansible
    hosts: hypervisor
    ansible_playbook:
      name: ansible/uninstall_ocp_libvirt.yml
    ansible_options:
      extra_vars:
        foo: bar
    ansible_galaxy_options:
      role_file: requirements.yml
    labels:
    - stack-beaker-libvirt-only
  labels:
  - stack-beaker-libvirt-only
```

**Comment by dno-github** [comment]
Jira ticket CCITCARBON-429 has been created for this issue. Url : https://issues.redhat.com/browse/CCITCARBON-429

**Comment by jbpratt** [comment]
Hi @rujutashinde this is still causing us issues, do you have any update on it?

**Comment by rujutashinde** [comment]
@jbpratt @ryankwilliams  I will take a look at this today/tomorrow get back to you soon. I will work on this in the coming week and provide a fix. I apologize for the delay.

**Comment by rujutashinde** [comment]
fixed with #211
